Source/WebCore/dom/ToggleEvent.cpp

```diff
@@ -66,4 +68,20 @@ Ref<ToggleEvent> ToggleEvent::createForBindings()
     return adoptRef(*new ToggleEvent);
 }
 
+RefPtr<Element> ToggleEvent::source() const
+{
+    if (!m_source)
+        return nullptr;
+
+    if (RefPtr target = dynamicDowncast<Node>(currentTarget())) {
+        Ref treeScope = target->treeScope();
+        Ref node = treeScope->retargetToScope(*m_source.get());
+        return &downcast<Element>(node).get();
```

Comment:
I think this should work:
```suggestion
        return downcast<Element>(node);
```
And if so, please also apply below.

Reply:
error: taking the address of a temporary object of type 'Ref<match_constness_t<Node, Element>>'

Looks like it needs the .get() it's a Ref<Eelement> * without it.

Reply:
Okay, in that case `downcast<Element>(node).ptr()` would be slightly better I suppose, but it still seems weird to me. Also, does that mean this can return `Element*` instead of `RefPtr<Element>`? That's generally preferable for non-allocating methods.